### PR TITLE
Add obs dockerfile and spec

### PR DESCRIPTION
**Pull request by davidcassany**
This PR adds the spec file and Dockerfile for `elemental` package and `node-image` image respectively. 

In addition it also introduces a new OBS workflow to build the package and image in a branch.

Part of #577

### CODE REVIEW

**Comment by davidcassany** [comment]
Note these are added under the `.obs/dockerfile` and `.obs/specfile` subfolders. Finally I did not use a `dist/obs` folder because I realized there was already an `.obs` folder for the OBS SCM integration, so I thought of reusing this one just to include all OBS files in a single place.

Probably instead of `.obs/specfile` and `.obs/dockerfile` there could be something like `.obs/recipefiles`?

**Review by Itxaka — APPROVED** [review]

File: .obs/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Open Build Service integration files
+
+This folder includes specific files to integrate with the Open Build Service (OBS) by
```
Comment (kkaempf) [inline .obs/README.md:3]:
```suggestion
This folder includes specific files to integrate with the [Open Build Service](https://build.opensuse.org) (OBS) by
```

File: .obs/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+openSUSE. Includes the workflows definition file (`workflows.yml`) that defines
```
Comment (kkaempf) [inline .obs/README.md:4]:
```suggestion
[openSUSE](https://www.opensuse.org). Includes the workflows definition file (`workflows.yml`) that defines
```

File: .obs/dockerfile/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: Apache-2.0
+# Define the names/tags of the container
+#!BuildTag: rancher/elemental-teal/5.3:latest
+#!BuildTag: rancher/elemental-teal/5.3:%VERSION%
+#!BuildTag: rancher/elemental-teal/5.3:%VERSION%-%RELEASE%
+#
+
+FROM suse/sle-micro-rancher/5.3:latest
```
Comment (kkaempf) [inline .obs/dockerfile/Dockerfile:8]: I wonder if we shouldn't make `5.3` a build config macro 🤔

**Review by kkaempf — CHANGES_REQUESTED** [review]

**Review by kkaempf — APPROVED** [review]

File: .obs/dockerfile/Dockerfile (hunk 3 above)
Comment (davidcassany) [reply .obs/dockerfile/Dockerfile:8]: Probably yes, but I would leave it for the time we need it. Hard to foresee now how it actually should be used.

**Pull request merged**